File: app/src/main/java/demo/victormunoz/githubusers/utils/picasso/ImageLoader.java
package demo.victormunoz.githubusers.utils.picasso;

import android.support.annotation.NonNull;
import android.widget.ImageView;

import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

public class ImageLoader {

    private final static String TAG = ImageLoader.class.getSimpleName();
    private final CircleTransformation transformation;
    private final Picasso picasso;

    public ImageLoader(Picasso picasso, CircleTransformation transformation){
        this.transformation = transformation;
        this.picasso = picasso;
    }

    public void load(@NonNull String url, @NonNull ImageView view){
        picasso.load(url).fit().tag(TAG).centerCrop().transform(transformation).into(view);
    }

    public void load(@NonNull String url, @NonNull ImageView view, @NonNull Callback callback){
        picasso.load(url).fit().tag(TAG).centerCrop().transform(transformation).into(view, callback);
    }

    public void cancelAll(){
        picasso.cancelTag(TAG);
    }

    public void cancelRequest(ImageView holder){
        picasso.cancelRequest(holder);
    }
}

File: app/src/main/java/demo/victormunoz/githubusers/utils/picasso/ImageLoader.kt
package demo.victormunoz.githubusers.utils.picasso

import android.widget.ImageView

import com.squareup.picasso.Callback
import com.squareup.picasso.Picasso

class ImageLoader(private val picasso: Picasso, private val transformation: CircleTransformation) {
    
    fun load(url: String, view: ImageView, callback: Callback) {
        picasso.load(url).fit().tag(TAG).centerCrop().transform(transformation).into(view, callback)
    }

    fun cancelAll() {
        picasso.cancelTag(TAG)
    }

    fun cancelRequest(holder: ImageView) {
        picasso.cancelRequest(holder)
    }

    companion object {

        private val TAG = ImageLoader::class.java.simpleName
    }
}
